js/views/conversation_search_view.js

```diff
@@ -84,6 +84,11 @@
             this.typeahead_view.collection.reset(
               this.typeahead.filter(isSearchable)
             );
+
+            // Check if the query is in the model list
+            // If it is then hide the new contact view
+            const modelExists = this.typeahead_view.collection.find(item => item.get('id') == query);
+            if (modelExists) this.new_contact_view.$el.hide();
```

Comment:
Good job finding the culprit!
Although it feels like there is a slightly better flow for this.
We're basically rendering a `new_contact_view` then later in the same function we're hiding it because the query string "may be a number" (Signal handles phone number, so read it as "may be a pub key").

Wouldn't it be more natural to first check if the query matches an existing contact pubkey, and only then show the typeahead results if not?

Reply:
I was trying to do that at first but it's not possible because searching in typeahead is async. This means if we want to change it to what you suggested, we need to only show the `new_contact_view` once we have search results. It's possible but in-terms of UX, it's better to allow the user to start the conversation before search results are returned.

Reply:
Makes sense!